dp189/components.py

```diff
@@ -106,3 +110,392 @@ def click_transactions(self):
+        """Click on the sorting drop-down and on the 'Model (A - Z)' option.
+
+        :return: None
+        """
+        self.click_drop_down()
+        self._model_a_z.click()
+
+    def click_model_z_a(self) -> None:
+        """Click on the sorting drop-down and on the 'Model (Z - A)' option.
+
+        :return: None
+        """
+        self.click_drop_down()
+        self._model_z_a.click()
+
+
+class ShowNumberProductsDropdownComponent:
+    """A drop-down list with number of displayed products options."""
+
+    def __init__(self, driver) -> None:
+        """All options in a drop-down list.
+
+        :param driver: Driver.
+        """
+        self._driver = driver
+        self._drop_down = driver.find_element(*LocatorShowNumberProducts.DROP_DOWN)
+        self._fifteen = driver.find_element(*LocatorShowNumberProducts.FIFTEEN)
+        self._twenty_five = driver.find_element(*LocatorShowNumberProducts.TWENTY_FIVE)
+        self._fifty = driver.find_element(*LocatorShowNumberProducts.FIFTY)
+        self._seventy_five = driver.find_element(*LocatorShowNumberProducts.SEVENTY_FIVE)
+        self._hundred = driver.find_element(*LocatorShowNumberProducts.HUNDRED)
+
+    def click_drop_down(self) -> None:
+        """Click on the drop-down.
+
+        :return: None
+        """
+        self._drop_down.click()
+
+    def click_fifteen(self) -> None:
+        """Click on the sorting drop-down and on the '15' option.
+
+        :return: None
+        """
+        self.click_drop_down()
+        self._fifteen.click()
+
+    def click_twenty_five(self) -> None:
+        """Click on the sorting drop-down and on the '25' option.
+
+        :return: None
+        """
+        self.click_drop_down()
+        self._twenty_five.click()
+
+    def click_fifty(self) -> None:
+        """Click on the sorting drop-down and on the '50' option.
+
+        :return: None
+        """
+        self.click_drop_down()
+        self._fifty.click()
+
+    def click_seventy_five(self) -> None:
+        """Click on the sorting drop-down and on the '75' option.
+
+        :return: None
+        """
+        self.click_drop_down()
+        self._seventy_five.click()
+
+    def click_hundred(self) -> None:
+        """Click on the sorting drop-down and on the '100' option.
+
+        :return: None
+        """
+        self.click_drop_down()
+        self._hundred.click()
+
+
+class ProductCompareLinkComponent:
+    """A link to compare the products added to the comparison."""
+
+    def __init__(self, driver) -> None:
+        """Initialise the link.
+
+        :param driver: Driver.
+        """
+        self._driver = driver
+        self._product_compare_link = driver.find_element(*LocatorProductCompareLink.PRODUCT_COMPARE)
+
+    def click_product_compare_link(self) -> None:
+        """Click on the 'Product Compare' link.
+
+        :return: None
+        """
+        self._product_compare_link.click()
+
+
+class ListViewComponent:
```

Comment:
In my opinion these grid and list elements are not finished yet. You can remove them for now and finish later.

Reply:
What else needs to be done for these elements?